implementation/influx/src/test/java/net/opentsdb/data/influx/TestInfluxLineProtocolParser.java

```diff
@@ -0,0 +1,817 @@
+// This file is part of OpenTSDB.
+// Copyright (C) 2020  The OpenTSDB Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package net.opentsdb.data.influx;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.when;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import net.opentsdb.common.Const;
+import net.opentsdb.data.LowLevelMetricData.ValueFormat;
+import net.opentsdb.utils.DateTime;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ DateTime.class })
+public class TestInfluxLineProtocolParser {
```

Comment:
I did the reverse job on this for our internal agent. Instead of decoding, I was concerned mostly with encoding. The test cases may give you more ideas about what could go wrong (or not).